File: Basics/1. Prime_Number.cpp
#include <bits/stdc++.h>
using namespace std;

    bool is_prime(int number){
        if(number<2)
        return false;
        else if(number==2)
        return true;
        else{
            for(int i = 3; i<=sqrt(number);i+=2){
                if(number%i==0)
                return false;
            }
            return true;
        }
    }
    int main(){
        int number;
        cout<<"enter number: ";
        cin>>number;
        if(is_prime(number))
        cout<<"Number "<<number<<" is prime";
        else
        cout<<"Number "<<number<<" is not prime";

        return 0;
}

File: Basics/1. Prime_Number.py
import math

def is_prime_pythonic(number):
    if number < 2:
        return False
    elif number == 2:
        return True
    else:
        for i in range(3, int(math.sqrt(number)) + 1, 2):
            if number % i == 0:
                return False
        return True

if __name__ == "__main__":
    num = int(input("Enter a number: "))
    if is_prime_pythonic(num):
        print(f"Number {num} is prime")
    else:
        print(f"Number {num} is not prime")